lib/service/database.js

```diff
@@ -418,6 +449,12 @@ function setCacheQueryTimeoutFromLimits(node, limits) {
     });
 }
 
+function getSourceNodes(node) {
```

Comment:
I thought the same but talking with Raul told me that is better to avoid adding specific cases and use the `getAllInputNodes` and make your own filter.

If we think is useful enough we could add it as you said